### PR TITLE
[Experimental] ParallelExecutor implemented by Thread-Cache queues

### PR DESCRIPTION
**Pull request by sneaxiy**
Test results on Resnet50 and Transformer:

85c286fdc335d1e3bef019d6520b8d25

### CODE REVIEW

File: paddle/fluid/framework/details/execution_strategy.h [hunk 1]
```diff
@@ -19,12 +19,13 @@ namespace framework {
 namespace details {
 
 struct ExecutionStrategy {
-  enum ExecutorType { kDefault = 0, kExperimental = 1 };
+  enum ExecutorType { kDefault = 0, kExperimental = 1, kDoubleQueue = 2 };
```
Comment (panyx0718) [inline paddle/fluid/framework/details/execution_strategy.h:22]: Document the difference of various strategy?

**Review by panyx0718 — COMMENTED** [review]
Maybe test it in CE and generate a report of performance difference?

**Comment by jacquesqiao** [comment]
Need some document, what is the purpose and optimization of these two kinds of executors?

File: paddle/fluid/framework/details/execution_strategy.h (hunk 1 above)
Comment (chengduoZH) [reply paddle/fluid/framework/details/execution_strategy.h:22]: Do you make speed comparison on some model?

Comment (sneaxiy) [reply paddle/fluid/framework/details/execution_strategy.h:22]: Yes, I would test it in several models.

**Comment by sneaxiy** [comment]
> Need some document, what is the purpose and optimization of these two kinds of executors?

I would add it soon after tested.

**Pull request closed without merging**